src/test/optimizer/strategy/subquery_to_join_rule_test.cpp

```diff
@@ -0,0 +1,1048 @@
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/limit_node.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "optimizer/strategy/subquery_to_join_rule.hpp"
+#include "storage/storage_manager.hpp"
+#include "utils/load_table.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+class SubqueryToJoinRuleTest : public StrategyBaseTest {
+ public:
+  void SetUp() override {
+    node_a = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "a");
+    a_a = node_a->get_column("a");
+    a_b = node_a->get_column("b");
+    a_c = node_a->get_column("c");
+    a_a_expression = to_expression(a_a);
+    a_b_expression = to_expression(a_b);
+    a_c_expression = to_expression(a_c);
+
+    node_b = MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}}, "b");
+    b_a = node_b->get_column("a");
+    b_b = node_b->get_column("b");
+
+    node_c = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "c");
+    c_a = node_c->get_column("a");
+    c_b = node_c->get_column("b");
+
+    node_d = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "d");
+    d_a = node_d->get_column("a");
+    d_b = node_d->get_column("b");
+    d_c = node_d->get_column("c");
+
+    node_e = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "e");
+    e_a = node_e->get_column("a");
+    e_b = node_e->get_column("b");
+    e_c = node_e->get_column("c");
+
+    _rule = std::make_shared<SubqueryToJoinRule>();
+  }
+
+  std::shared_ptr<SubqueryToJoinRule> _rule;
+
+  std::shared_ptr<MockNode> node_a, node_b, node_c, node_d, node_e;
+  LQPColumnReference a_a, a_b, a_c, b_a, b_b, c_a, c_b, d_a, d_b, d_c, e_a, e_b, e_c;
+  std::shared_ptr<LQPColumnExpression> a_a_expression, a_b_expression, a_c_expression;
+};
+
+// HELPER FUNCTION TESTS
+
+TEST_F(SubqueryToJoinRuleTest, AssessCorrelatedParameterUsageCountsNodesNotUsages) {
+  const auto parameter1 = correlated_parameter_(ParameterID{0}, a_a);
+  const auto parameter2 = correlated_parameter_(ParameterID{1}, a_b);
+  const std::map<ParameterID, std::shared_ptr<AbstractExpression>> parameter_map = {{ParameterID{0}, a_a_expression},
+                                                                                    {ParameterID{1}, a_b_expression}};
+
+  // clang-format off
+  const auto lqp =
```

Comment:
@Felerius don't know how WIP this is, so sorry if you wanted to change this anyway: please format LQPs in the tests with an indentation of two.

Reply:
They must have gotten accidentally auto-formatted at one point. Fixed now, thanks!